src/EditorFeatures/Test/CodeActions/CodeChangeProviderMetadataTests.cs

```diff
@@ -0,0 +1,188 @@
+{
+    [UseExportProvider]
+    public class CodeChangeProviderMetadataTests
+    {
+        [Theory]
+        [InlineData(typeof(CodeFixProvider))]
+        [InlineData(typeof(CodeRefactoringProvider))]
+        [InlineData(typeof(IConfigurationFixProvider))]
+        public void TestNameMetadataIsPresent(Type providerType)
+        {
+            var configuration = EditorTestCompositions.EditorFeatures.GetCompositionConfiguration();
+            var exportedProviders = FindComposedPartsWithExport(configuration, providerType.FullName).ToArray();
+
+            var failureMessage = new StringBuilder();
+            failureMessage.AppendLine($"The following {providerType.Name}s exported without Name metadata:");
+            var passLength = failureMessage.Length;
+
+            foreach (var (providerPart, providerExport) in exportedProviders)
+            {
+                if (!providerExport.Metadata.TryGetValue("Name", out var name)
+                    || name is not string { Length: > 0 })
+                {
+                    failureMessage.AppendLine(providerPart.Definition.Type.FullName);
+                }
+            }
+
+            Assert.True(failureMessage.Length == passLength, failureMessage.ToString());
+        }
+
+        [Theory]
+        [InlineData(typeof(CodeFixProvider), LanguageNames.CSharp)]
+        [InlineData(typeof(CodeFixProvider), LanguageNames.VisualBasic)]
+        [InlineData(typeof(CodeRefactoringProvider), LanguageNames.CSharp)]
+        [InlineData(typeof(CodeRefactoringProvider), LanguageNames.VisualBasic)]
+        [InlineData(typeof(IConfigurationFixProvider), LanguageNames.CSharp)]
+        [InlineData(typeof(IConfigurationFixProvider), LanguageNames.VisualBasic)]
+        public void TestNameMetadataIsUniqueAmongProviders(Type providerType, string language)
+        {
+            var configuration = EditorTestCompositions.EditorFeatures.GetCompositionConfiguration();
+            var exportedProviders = FindComposedPartsWithExportForLanguage(configuration, providerType.FullName, language);
+
+            var failureMessage = new StringBuilder();
+            failureMessage.AppendLine($"The following {providerType.Name}s are exported for {language} without unique Name metadata:");
+            var passLength = failureMessage.Length;
+
+            var exportedProvidersByName = exportedProviders.GroupBy(exportedProvider =>
+            {
+                if (!exportedProvider.Export.Metadata.TryGetValue("Name", out var name)
+                    || name is not string { Length: > 0 })
+                {
+                    return string.Empty;
+                }
+
+                return (string)name;
+            });
+
+            foreach (var namedGroup in exportedProvidersByName)
+            {
+                if (string.IsNullOrEmpty(namedGroup.Key))
+                {
+                    continue;
+                }
+
+                if (namedGroup.Take(2).Count() == 1)
```

Comment:
huh?

Reply:
What? I wanted to know if there were more than 2 items, and I didn't want to enumerate the complete list. =)

Reply:
I'll change to just use Count().